api/src/main/java/org/openmrs/CohortMembership.java

```diff
@@ -0,0 +1,104 @@
+/**
+ * This Source Code Form is subject to the terms of the Mozilla Public License,
+ * v. 2.0. If a copy of the MPL was not distributed with this file, You can
+ * obtain one at http://mozilla.org/MPL/2.0/. OpenMRS is also distributed under
+ * the terms of the Healthcare Disclaimer located at http://openmrs.org/license.
+ *
+ * Copyright (C) OpenMRS Inc. OpenMRS is a registered trademark and the OpenMRS
+ * graphic logo is a trademark of OpenMRS Inc.
+ */
+package org.openmrs;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+
+import java.util.Date;
+
+public class CohortMembership extends BaseOpenmrsData implements Comparable<CohortMembership> {
+	
+	public static final long serialVersionUID = 0L;
+	
+	protected static final Log log = LogFactory.getLog(CohortMembership.class);
+
+	private Integer cohortMemberId;
+	
+	private Cohort cohort;
+	
+	private Patient patient;
+	
+	private Date startDate;
+	
+	private Date endDate;
+
+	// Constructor
+	public CohortMembership() {
+	}
+
+	public CohortMembership(Patient patient, Date startDate) {
+		this.patient = patient;
+		this.startDate = startDate;
+	}
+
+	public CohortMembership(Patient patient) {
+		this(patient, new Date());
+	}
+	
+	public boolean isMemberActive() {
+		return this.getStartDate().before(new Date()) && this.getEndDate() == null;
```

Comment:
startDate has not-null constraint. The changeset has the insert to ensure existing records have the startDate.